attribute/genericattribute.go

```diff
@@ -11,8 +11,10 @@ import (
 //GenericAttribute is a Yoti attribute which returns a generic value
 type GenericAttribute struct {
 	*yotiprotoattr_v3.Attribute
-	Value   interface{}
-	Anchors []*anchor.Anchor
+	Value     interface{}
+	Anchors   []*anchor.Anchor
```

Comment:
Are these lists immutable?

Reply:
Doesn't look like it's currently possible to add readonly modifiers from these proposals [[1]](https://github.com/golang/go/issues/20443) [[2]](https://github.com/golang/go/issues/22876)

We could remove them as fields on the struct, and instead expose GetAnchors, GetSources and GetVerifier methods. I did consider this, but it would mean quite a bit of duplication as we would need to define this on each of the different attribute types in the current way it's set up.

Reply:
This is a shame.
But we do to expose immutable objects from the SDKs, somehow...